kernel/contracts/ZepCore.sol

```diff
@@ -0,0 +1,85 @@
+pragma solidity ^0.4.18;
+
+import "./ZepToken.sol";
+import "./KernelInstance.sol";
+import "./KernelRegistry.sol";
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract ZepCore {
+  using SafeMath for uint256;
+
+  event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
+  event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
+
+  uint256 public newVersionCost;
+  uint256 public developerFraction;
+
+  ZepToken private _token;
+  KernelRegistry private _registry;
+  uint256 private _totalStaked;
+  mapping(address => uint256) private _instanceVouches;
+  mapping(address => mapping (address => uint256)) private _userVouches;
+
+  function ZepCore(uint256 _newVersionCost, uint256 _developerFraction) public {
+    _token = new ZepToken();
+    _registry = new KernelRegistry();
+    developerFraction = _developerFraction;
+    newVersionCost = _newVersionCost;
+    // TODO: we need to think how we are going to manage variable costs to propose new versions
+  }
+
+  function register(KernelInstance instance) public {
+    registry().addInstance(instance);
+    token().transferFrom(msg.sender, this, newVersionCost);
+    token().burn(newVersionCost);
+    // TODO: should we build a burnFrom or sth like that?
```

Comment:
I'd rather keep a standard interface, I like how it is implemented now

Reply:
Ok, but keep in mind that the current approach requires allowances to be set for ZepCore prior to usage, which at some point we will want to encapsulate to improve usability..

Reply:
I'd push for unlimited allowances in OZ :-P